### PR TITLE
add support for Accessors, patch a weird type issue

### LINKED ISSUE #32: Safely modify a constraint tree
At the moment we have mutable constraint trees, which is very handy for manipulating complex structures, e.g. changing bounds etc. The problem is that this is a common pattern:
```julia
wt = simplified_enzyme_constrained_flux_balance_constraints(
        model;
        reaction_isozymes,
        gene_product_molar_masses,
        capacity = 400.0, # mg/gDW
        interface = :identifier_prefixes,
)
 wt.fluxes[rxn_id].bound = C.Between(-1000, 1000)
```
but it won't always work, because you also need to do:
```julia
wt.fluxes_forward[rxn_id].bound = C.Between(0, 1000)
wt.fluxes_reverse[rxn_id].bound = C.Between(0, 1000)
wt.interface.exchanges[rxn_id].bound = C.Between(-1000, 1000)
```
This makes is really easy to get incorrect results results if you don't know all the places the original flux bound is also implicitly present...

### PR DESCRIPTION
**Pull request by exaexa**
Closes #32

### CODE REVIEW

File: docs/src/1-metabolic-modeling.jl [hunk 1]
```diff
@@ -391,6 +392,39 @@ result.exchanges
 
 @test result_with_more_oxygen.exchanges.oxygen < -19.0 #src
 
+# ### Alternative: Using Accessors.jl
+#
+# Accessors.jl implement a "lensy" way to update immutable data structures.
+# That comes with a nice outcome of doing the right amount of shallow copyies
+# for you automatically, thus avoiding much of the technical danger of in-place
+# modifiations. (You still lose the equational reasoning on your code, but that
```
Comment (stelmo) [inline docs/src/1-metabolic-modeling.jl:400]: modifications

File: docs/src/1-metabolic-modeling.jl [hunk 2]
```diff
@@ -391,6 +392,39 @@ result.exchanges
...
+# nested field in a structure, returning a modified copy), or with `@reset`
+# which automatically "assigns" the result back to the original variable:
+
+using Accessors
+
+c = @set c.exchanges.biomass.bound = C.Between(-50, 50)
+
+# The above code is equivalent to:
+
+@reset c.exchanges.biomass.bound = C.Between(-50, 50)
```
Comment (stelmo) [inline docs/src/1-metabolic-modeling.jl:413]: Unclear what the difference is

**Review by stelmo — APPROVED** [review]

File: docs/src/1-metabolic-modeling.jl (hunk 1 above)
Comment (exaexa) [inline docs/src/1-metabolic-modeling.jl:400]:
```suggestion
# modifications. (You still lose the equational reasoning on your code, but that
```

File: docs/src/1-metabolic-modeling.jl (hunk 2 above)
Comment (exaexa) [reply docs/src/1-metabolic-modeling.jl:413]: it's described 5 lines above, basically "you don't have to `c=` manually".

Should I copypaste or move that here?

Comment (exaexa) [reply docs/src/1-metabolic-modeling.jl:413]: @stelmo ping ping

Comment (stelmo) [reply docs/src/1-metabolic-modeling.jl:413]: Ah yeah, on a second reading it was clear :)

**Pull request merged**